wave_kinematics/tests/rotation_test.cpp

```diff
@@ -0,0 +1,185 @@
+#include <iostream>
+#include <random>
+
+#include <gtest/gtest.h>
+
+#include "wave/kinematics/numerical_test_functors.hpp"
+#include "wave/kinematics/rotation.hpp"
+#include "wave/utils/math.hpp"
+#include "wave/utils/utils.hpp"
+
+
+class RotationTestFixture : public ::testing::Test {
+public:
+  // Data members used for testing.
+  wave::Rotation rotation1;
+  wave::Rotation rotation1_truth;
+  wave::Rotation rotation2;
+  wave::Rotation rotation3;
+  wave::Rotation rotation1_inverse;
+  wave::Rotation eye3;
+
+  // A vector v1.
+  wave::Vec3 v1;
+  // v1 rotated by rotation1.
+  wave::Vec3 v1_rotated;
+  // True angle axis params for rotation1.
+  wave::Vec3 rotation1_angleaxis_params;
+
+  void SetUp() {
+    wave::Rotation::fromEulerXYZ(0.1, 0.2, 0.3, rotation1);
```

Comment:
Most of the stuff in `RotationTestFixture` would be better placed in the individual tests which check those functions.  A test fixture makes sense if you have to generate a bunch of input data that comes from _not_ the functions under test. A single result _from_ the FUT does not belong in the fixture.

See http://blog.stevensanderson.com/2009/08/24/writing-great-unit-tests-best-and-worst-practises/ 
> Avoid having common setup code that runs at the beginning of lots of unrelated tests. Otherwise, it’s unclear what assumptions each test relies on, and indicates that you’re not testing just a single unit.